src/Extension/ExtensionManager.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Carnival\Extension;
+
+use Carnival\Extension\Contracts\ExtensionManager as ExtensionManagerContract;
+use Carnival\Extension\Contracts\Extension;
+use Carnival\Extension\Contracts\Installable;
+use Illuminate\Support\Collection;
+use InvalidArgumentException;
+
+class ExtensionManager implements ExtensionManagerContract {
+    private Collection $extensions;
+
+    public function __construct() {
+        $this->extensions = new Collection();
```

Comment:
I think I mentioned this before, but is there a difference between a new Collection() or collect()

Reply:
`collect()` is a helper method created in the [helpers.php](https://github.com/illuminate/collections/blob/master/helpers.php) file. That file doesn't create a class, though, it's meant to be used by an [autoloader](https://stackoverflow.com/a/28290359) so it's always in scope. But we're not autoloading the file for Carnival since it'll be autoloaded in AKFS ([by Laravel](https://github.com/laravel/framework/blob/10.x/src/Illuminate/Collections/helpers.php#L16))

Reply:
Thanks for the explanation!

Reply:
Actually, I didn't realize that the `illuminate/collections` repo autoloads it itself as well: https://github.com/illuminate/collections/blob/master/composer.json#L27

So `collect()` will work here as well. I'll fix

Reply:
oh nice, sounds good then.